pkg/k8s/client.go

```diff
@@ -81,23 +81,28 @@ func NewKubeClient() (Client, error) {
 	return kubeClient, nil
 }
 
-// GetPodName get pod ns,name from apiserver
-func (c *KubeClient) GetPodName(podip string) (podns, poddname string, err error) {
+// GetPodInfo get pod ns,name from apiserver
+func (c *KubeClient) GetPodInfo(podip string) (podns, poddname, deployment string, err error) {
 	if podip == "" {
-		return "", "", fmt.Errorf("podip is empty")
+		return "", "", "", fmt.Errorf("podip is empty")
 	}
 
 	podList, err := c.getPodListWithTries(podip, getPodListTries, getPodListSleepTimeMilliseconds)
 
 	if err != nil {
-		return "", "", err
+		return "", "", "", err
 	}
 	numMatching := len(podList.Items)
 	if numMatching == 1 {
-		return podList.Items[0].Namespace, podList.Items[0].Name, nil
+		// TODO: Filter out the deployment owner references.
+		deployment := ""
+		if podList.Items[0].OwnerReferences[0].Kind == "ReplicaSet" {
+			deployment = podList.Items[0].OwnerReferences[0].Name
```

Comment:
This is the replicaset name, so maybe we should change deployment to rsName to reflect the value we are getting